sql/src/main/codegen/includes/create-source.ftl

```diff
@@ -0,0 +1,36 @@
+SqlCreateSource SqlCreateSource(Span s, boolean replace) :
+{
+    final SqlIdentifier id;
+    final Map<SqlNode, SqlNode> map;
+}
+{
+    <SOURCE> id = SimpleIdentifier()
+    <WITH> <LPAREN>
+    map = Properties()
+    <RPAREN>
+    {
+      return new SqlCreateSource(s.end(this), id, map);
+    }
+}
+
+Map<SqlNode, SqlNode> Properties() :
+{
+    final Map<SqlNode, SqlNode> props = new HashMap<SqlNode, SqlNode>();
+    SqlNode key;
+    SqlNode value;
+}
+{
+    key = StringLiteral() <EQ> value = StringLiteral()
+    {
+        props.put(key, value);
+    }
+    (
+        <COMMA> key = StringLiteral() <EQ> value = StringLiteral()
```

Comment:
Should we allow values that are not string literals (such as numbers)?

Reply:
Also, just for ergonomics (I know this deviates from the design docs a bit), what do you think about:

- Making the parentheses around the properties optional
- Allowing keys without quotes around them (as long as they don't contain a space) – Snowflake does that
- Allowing an optional comma after the last pair in the list 

For example, all these would be equivalent:
```sql
-- Variant 1
CREATE SOURCE data
WITH (
  'connector' = 's3',
  'path' = 's3://mybucket/file.csv',
);

-- Variant 2
CREATE SOURCE data
WITH (
  connector = 's3',
  path = 's3://mybucket/file.csv',
);

-- Variant 3
CREATE SOURCE data
WITH
  connector = 's3',
  path = 's3://mybucket/file.csv',
;

-- Variant 4
CREATE SOURCE data WITH connector = 's3', path = 's3://mybucket/file.csv';
```

If it's hard or you don't like it, nevermind, we can return to the discussion at another time :)

Reply:
Loved the proposed ergonomic improvements. If its easy lets get this in. :)

Reply:
nice suggestions, I can look into adding whatever are straightforward to implement. Also StringLiteral() will allow numbers.